src/v/cloud_storage/remote.cc

```diff
@@ -721,7 +721,7 @@ cloud_roles::credentials auth_refresh_bg_op::build_static_credentials() const {
       [](const auto& cfg) {
           using cfg_type = std::decay_t<decltype(cfg)>;
           if constexpr (std::is_same_v<
-                          cloud_storage_clients::configuration,
```

Comment:
soud it be `cloud_storage_clients::tiered_storage_configuration` `s3` being just one impl?

Reply:
That's the idea. `cloud_storage_clients::client_configuration` is a sum type that can house the different configs for S3 and ABS: `cloud_storage_clients::s3_configuration` and `cloud_storage_clients::abs_configuration` (added in the follow up PR https://github.com/VladLazar/redpanda/pull/4/commits/26fc1543c91a331d494924b5f34e177407978133#diff-4e4f592de067010cdae80e75507d9c492bddab7f1cde7ec54511e932bfb606bb).

Are you suggesting to nest the configs in a namesapce: `cloud_storage_clients::configuration::s3` and `cloud_storage_clients::configuration::abs`?